src/Serializer.php

```diff
@@ -118,8 +133,172 @@ public static function decodeMetadata($metadata)
         return $result;
     }
 
+    private function encodeElement(FieldDescriptor $field, $data)
```

Comment:
This is for encoding/decoding to and from PHP arrays, rather than a wire format, which protobuf doesn't support. This is necessary for APIs like Pubsub, Logging and Spanner with manual veneers. Previously this was being handled in google-cloud-php, but it is general functionality that I wanted to pull into GAX because it could be useful to GAPICs in future.